### ISSUE TITLE
check_callable import raises on python 3.5.1+

### ISSUE THREAD

**Issue by ether42** [agronholm/typeguard#2]
I'm new to the type annotation system introduced in Python 3 and I tried to start a simple project (using asphalt, which in turns uses typeguard) on Debian stretch and it will traceback when importing from typeguard.
I also tried to compile the provided Python 3.5 & 3.6 tarballs from here: https://www.python.org/downloads/source/

```shell
python3.5 --version
Python 3.5.1+
```

The asphalt project is doing a simple
```python
from asphalt.core.main import *
```
Which will do
```python
from typeguard import check_argument_types
```
Which will provoke the following traceback:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File ".../tmpvenv/lib/python3.5/site-packages/typeguard.py", line 37, in <module>
    def check_callable(argname: str, value, expected_type, typevars_memo: Dict[TypeVar, type]) -> None:
  File "/usr/lib/python3.5/typing.py", line 1018, in __getitem__
    tvars = _type_vars(params)
  File "/usr/lib/python3.5/typing.py", line 282, in _type_vars
    _get_type_vars(types, tvars)
  File "/usr/lib/python3.5/typing.py", line 277, in _get_type_vars
    t._get_type_vars(tvars)
TypeError: _get_type_vars() missing 1 required positional argument: 'tvars'
```

However, it works as expected with a Python 3 on Ubuntu that I got from the ppa https://launchpad.net/~fkrull/+archive/ubuntu/deadsnakes (it seems there wasn't any `_get_type_vars` function in this version):
```shell
python3.5 --version
Python 3.5.1
```

Do you known if this is the expected behavior?

**Comment by agronholm** [comment]
Frankly I have no idea how that error would happen. Can you provide specific instructions for reproducing it? Doesn't happen with Ubuntu's stock Python 3.5.

**Comment by ether42** [comment]
Yup, as I said, simply download the Python tarball from the official site (copy/paste from shell history that I just did at your request on an Ubuntu machine):

```shell
wget https://www.python.org/ftp/python/3.5.2/Python-3.5.2rc1.tgz
tar xvf Python-3.5.2rc1.tgz
cd Python-3.5.2rc1
./configure --prefix=`pwd`/../python
make -j8
make install
cd ..
./python/bin/python3.5 -m venv venv
source venv/bin/activate
pip install typeguard
python -c 'from typeguard import check_callable'
```

Will give you the same traceback as above.

Thanks for the fast answer :)

**Comment by agronholm** [comment]
You left out the fact that the error only occurred on Python 3.5.2. I think that is the key point. Also, could you try the final release instead of the release candidate? Thanks.

**Comment by ether42** [comment]
Oh, my bad I didn't think too much about the fact that I used the release candidate (just took the latest version).

So, here are the results:
 - Python-3.5.1: ok
 - Python-3.5.2rc1: broken
 - Python-3.6.0a2: broken

If I diff between 3.5.1 & 3.5.2rc1 I get a similar difference between the Python 3.5.1 of Ubuntu and the Python 3.5.1+ from Debian Stretch.

So maybe some behavior will change in the future?

**Comment by agronholm** [comment]
But did you try with 3.5.2 final?

**Comment by ether42** [comment]
It is also failing with the same traceback.
(Sorrry, I don't follow that much Python releases, didn't even see there was a final above the release candidate...)

**Comment by agronholm** [comment]
It seems like the TypeVar is the problem.

**Comment by agronholm** [comment]
Fixed in 8a07cd6.

**Comment by agronholm** [comment]
I've released v1.2.1 that fixes the compatibility issues with 3.5.2.
